test/Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -153,9 +157,55 @@ spec = Test.group "Dataflow Warnings" <|
         mapped_3.stack_trace.second.name . should_equal "Warnings_Spec.throw_a_bar"
         Warning.get_all mapped_3 . catch . should_equal "bar"
 
+    Test.specify "should allow to detach warnings, selectively" <|
+        warned = attach_four_warnings "foo"
+        detached_pair = Warning.detach_selected_warnings warned odd_warning_filter
+        detached_pair . first . should_equal "foo"
+        Warning.get_all (detached_pair . first) . map .value . should_equal [2,4]
+        detached_pair.second . map .value . should_equal [1,3]
+
     Test.specify "should allow to map the warnings, selectively" <|
         x = Warning.attach "foo" 1
         result = x.is_static_nothing x
         result . should_equal False
 
+    Test.specify "should be allowed in Vector" <|
+        a = Warning.attach "a" 1
+        b = Warning.attach "b" 2
+        c = [a, b]
+        d = Warning.attach "d" c
+        c . should_equal [1, 2]
+        Warning.get_all c . map .value . should_equal ["b", "a"]
+        Warning.get_all d . map .value . should_equal ["d", "b", "a"]
+
+    Test.specify "should be preserved after operations on Vector" <|
+        even x = (Warning.attach x x) % 2
+        res = [1,2,3,4].map even
+
+        res . should_equal [1, 0, 1, 0]
+        Warning.get_all res . map .value . should_equal [4, 3, 2, 1]
+        res.at 0 . should_equal 1
+        Warning.get_all (res.at 0) . map .value . should_equal [1]
```

Comment:
Tbh I don't understand this logic.

How if `res` has warnings `[4, 3, 2, 1]`, the call to `res.at 0` has only `[1]`?

To be consistent with the rest of our logic I think the mere fact that we are calling _something_ on `res` means the result should get 'infected' with _all_ warnings of `res`.

I know that in case of arrays it may seem more intuitive to just return the warnings associated with the stored value. But this is inconsistent with other of our primitives like atoms - see for example:
![image](https://user-images.githubusercontent.com/1436948/206010993-935c5cfe-17a8-4508-a433-27c02fe0c335.png)

Reply:
In reality, it actually depends on the actual _meaning_ of a given warning whether it should infect the whole collection or just keep itself to a single element and only show up in whole collection accesses.

Still, if we are going for 'stick warnings with elements instead of whole array', what about the test below?

```diff
         slice = res.slice 1 4
-        Warning.get_all slice . map .value . should_equal [4, 3, 2, 1]
+       Warning.get_all slice . map .value . should_equal [4, 3, 2]
```

IMO we should change it to the following to reflect that semantics. Of course this complicates slicing quite a lot...

I just want to demonstrate that if we want this to behave in a consistent way, making `res.at 0` return _just_ warnings of the first element has some complicated implications.

Reply:
Finally, I'm not against strongly against `res.at 0` returning _just_ `[1]`. But then, shouldn't in my REPL example above, `p.first` also only return `["a"]`?

Reply:
The reason why all this propagation of warnings happens for fields/methods is due to the invocation logic that explicitly extracts/propagates warnings for `self` arguments.
The reason why it doesn't work that way for Vector (and anything Array-like) is because accessing elements is completely done via interop. So it was on purpose. 

I was going to delay this until we have a separate discussion but since you brought it up now I picked up the slack. This is actually fairly to easy to implement now with `WarningsLibrary` as demonstrated in the last commit, although the whole warnings stuff pollutes a lot of the logic :shrug:

Reply:
Thanks!